backend/handlers/like_handler.py

```diff
@@ -44,54 +44,62 @@ def get(self, user, post_key, comment_id=None, reply_id=None):
 
     @json_response
     @login_required
-    @ndb.transactional(xg=True)
     def post(self, user, post_key, comment_id=None, reply_id=None):
         """Handle POST Requests."""
-        """This method is only meant to give like in post."""
-        post = ndb.Key(urlsafe=post_key).get()
-        institution = post.institution.get()
-        body = json.loads(self.request.body)
-        entity_type = 'LIKE_POST'
-        
-        Utils._assert(institution.state == 'inactive',
-                      "The institution has been deleted", NotAuthorizedException)
-
+        """This method is only meant to give like in post, comment or reply."""
         if comment_id:
-            comment = post.get_comment(comment_id)
-            entity_type = 'LIKE_COMMENT'
-            if reply_id:
-                comment = comment.get('replies').get(reply_id)
-
-            likes = comment.get('likes')
+            @ndb.transactional(retries=10)
+            def like_comment(post_key, user, comment_id=None, reply_id=None):
+                """Increment one 'like' in  comment or reply and send notification.""" 
+                post = ndb.Key(urlsafe=post_key).get()
+                comment = post.get_comment(comment_id)
+                if reply_id:
+                    comment = comment.get('replies').get(reply_id)
+                likes = comment.get('likes')
+                Utils._assert(user.key.urlsafe() in likes,
+                            "User already liked this comment", NotAuthorizedException)
+                likes.append(user.key.urlsafe())
```

Comment:
You should extract that list modification to the model, avoiding the handler to modify the entities.